src/components/Calculator.js

```diff
@@ -1,47 +1,56 @@
 import React, { Component } from 'react';
+import calculate from '../logic/calculate';
 
 class Calculator extends Component {
   constructor(props) {
     super(props);
-    this.state = {};
+    this.state = {
+      total: 0,
+      next: null,
+      operation: null,
+    };
+
+    this.handleClick = this.handleClick.bind(this);
+  }
+
+  handleClick(e) {
+    e.preventDefault();
+    this.setState((prev) => calculate(prev, e.target.name));
   }
 
   render() {
+    const { total, next, operation } = this.state;
+    const buttons = [['AC', '+/-', '%', '÷'], ['7', '8', '9', 'x'], ['4', '5', '6', '-'], ['1', '2', '3', '+']].map((row) => (
+      <div className="row" key={row[0]}>
+        <button type="button" className="col" id={row[0]} name={row[0]} onClick={this.handleClick}>{row[0]}</button>
+        <button type="button" className="col" id={row[1]} name={row[1]} onClick={this.handleClick}>{row[1]}</button>
+        <button type="button" className="col" id={row[2]} name={row[2]} onClick={this.handleClick}>{row[2]}</button>
+        <button type="button" className="col operature" id={row[3]} name={row[3]} onClick={this.handleClick}>{row[3]}</button>
+      </div>
```

Comment:
**Changes Required**
- [ ] Well done storing the button name in any array and using `array.map` to dynamically add then to the DOM. This can still be improved by refactoring the `Button` into it's own Class component that receives the `button name` and `onClick evnet` as props from its parent the calculator.  Let us try to refactor this into a `Button Component` and  pass the name and onClick event as `props`.

Reply:
Nice idea 💡 .. I add this *Button component* thank you for your feedback
_it's not in requirement but still good practice 👍🏼 _